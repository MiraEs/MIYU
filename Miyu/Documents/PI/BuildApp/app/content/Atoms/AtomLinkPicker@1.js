import React, {
	Component,
} from 'react';
import PropTypes from 'prop-types';
import {
	ViewPropTypes,
} from 'react-native';
import { TouchableOpacity } from 'BuildLibrary';
import { STORE_ID } from '../../constants/constants';
import { withNavigation } from '@expo/ex-navigation';
import {
	ARTICLE,
	CATEGORY,
	ROUTE,
} from '../../constants/ContentConstants';
import { connect } from 'react-redux';

@withNavigation
export class AtomLinkPicker extends Component {

	constructor(props) {
		super(props);
		const { links } = this.props;
		if (links && links[STORE_ID]) {
			const link = links[STORE_ID];
			if (link.linkType && link.data && Array.isArray(link.data.selected)) {
				const selected = link.data.selected[0];
				switch (link.linkType.toLowerCase()) {
					case 'category': {
						const categoryId = selected.categoryId;
						this.linkRoute = 'category';
						this.linkProps = {
							selectedFacets: link.facets ? link.facets.selected : [],
							storeId: selected.storeId,
							categoryId,
						};
						this.trackingData = { _c: categoryId };
						break;
					}
					case 'video': {
						this.linkRoute = 'video';
						const video = props.videoIncludes[selected];
						this.linkProps = {
							...video,
						};
						break;
					}
					case 'product': {
						this.linkRoute = 'productDetail';
						let uniqueId;
						if (link.finish) {
							if (typeof link.finish.uniqueId === 'number') {
								uniqueId = link.finish.uniqueId;
							}
						}
						this.linkProps = {
							compositeId: selected,
							uniqueId,
						};
						this.trackingData = { _s: selected };
						break;
					}
					case 'favorite': {
						this.linkRoute = 'favoritesList';
						this.linkProps = {
							favoriteId: selected,
						};
						this.trackingData = { _f: selected };
						break;
					}
					case 'article': {
						this.linkRoute = 'content';
						this.linkProps = {
							id: selected,
							type: ARTICLE,
						};
						this.trackingData = { _a: selected };
						break;
					}
					case 'page': {
						this.linkRoute = 'content';
						this.linkProps = {
							id: selected.id,
							type: ROUTE,
						};
						this.trackingData = { _r: selected.id };
						break;
					}
					case 'brand': {
						this.linkRoute = 'content';
						this.linkProps = {
							id: selected.id,
							type: CATEGORY,
						};
						this.trackingData = { _b: selected.id };
						break;
					}
					case 'profile': {
						this.linkRoute = 'profile';
						this.linkProps = {
							profile: props.profileIncludes[selected],
						};
					}
				}
			}
		}
	}

	render() {
		const link = this.props.links[STORE_ID];
		const { group } = this.props;
		if (link && link.data && this.linkRoute) {
			return (
				<TouchableOpacity
					onPress={() => {
						if (this.linkRoute) {
							this.props.navigator.push(this.linkRoute, this.linkProps);
						}
					}}
					trackAction={this.props.trackAction}
					trackContextData={{
						...this.trackingData,
						_cr: this.props.contentItemId,
						_g: group.id,
					}}
					activeOpacity={0.5}
				>
					{this.props.children}
				</TouchableOpacity>
			);
		} else {
			return null;
		}
	}

}

AtomLinkPicker.propTypes = {
	children: PropTypes.oneOfType([
		PropTypes.array,
		PropTypes.string,
		PropTypes.element,
		PropTypes.number,
	]).isRequired,
	group: PropTypes.object.isRequired,
	contentItemId: PropTypes.string.isRequired,
	links: PropTypes.object.isRequired,
	navigator: PropTypes.object,
	accentColor: PropTypes.bool,
	style: ViewPropTypes.style,
	trackAction: PropTypes.string.isRequired,
	profileIncludes: PropTypes.object,
	videoIncludes: PropTypes.object,
};

export default connect((state) => {
	return {
		profileIncludes: state.contentReducer.profileIncludes,
		videoIncludes: state.contentReducer.videoIncludes,
	};
}, null)(AtomLinkPicker);
