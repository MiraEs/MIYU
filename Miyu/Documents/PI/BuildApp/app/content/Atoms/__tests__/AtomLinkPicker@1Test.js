'use strict';

jest.mock('BuildNative');
jest.mock('../../../../app/services/httpClient', () => ({}));
jest.unmock('react-native');

import { AtomLinkPicker } from '../AtomLinkPicker@1';
import React from 'react';

const defaultProps = {
	actions: {},
	children: [],
	group: { id: 1 },
	contentItemId: 'contentItemId',
	links: {},
	trackAction: 'trackAction',
};

describe('AtomLinkPicker component', () => {
	it('should render correctly', () => {
		const tree = require('react-test-renderer').create(
			<AtomLinkPicker {...defaultProps} />
		).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should render correctly for category', () => {
		const links = {
			'248': {
				linkType: 'category',
				facets: [],
				data: {
					selected: [{ storeId: '248', categoryId: '123456' }],
				},
			},
		};

		const tree = require('react-test-renderer').create(
			<AtomLinkPicker
				{...defaultProps}
				links={links}
			/>
		).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should render correctly for video', () => {
		const links = {
			'248': {
				linkType: 'video',
				data: {
					selected: ['video'],
				},
			},
		};

		const videoIncludes = { 'video': { 'foo': 'bar' } };

		const tree = require('react-test-renderer').create(
			<AtomLinkPicker
				{...defaultProps}
				links={links}
				videoIncludes={videoIncludes}
			/>
		).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should render correctly for profile', () => {
		const links = {
			'248': {
				linkType: 'profile',
				data: {
					selected: ['profile'],
				},
			},
		};

		const profileIncludes = { 'profile': { 'foo': 'bar' } };

		const tree = require('react-test-renderer').create(
			<AtomLinkPicker
				{...defaultProps}
				links={links}
				profileIncludes={profileIncludes}
			/>
		).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should render correctly for product', () => {
		const links = {
			'248': {
				linkType: 'product',
				finish: 1234,
				data: {
					selected: ['12345'],
				},
			},
		};

		const tree = require('react-test-renderer').create(
			<AtomLinkPicker
				{...defaultProps}
				links={links}
			/>
		).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should render correctly for favorite', () => {
		const links = {
			'248': {
				linkType: 'favorite',
				data: {
					selected: ['12345'],
				},
			},
		};

		const tree = require('react-test-renderer').create(
			<AtomLinkPicker
				{...defaultProps}
				links={links}
			/>
		).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should render correctly for article', () => {
		const links = {
			'248': {
				linkType: 'article',
				data: {
					selected: ['12345'],
				},
			},
		};

		const tree = require('react-test-renderer').create(
			<AtomLinkPicker
				{...defaultProps}
				links={links}
			/>
		).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should render correctly for page', () => {
		const links = {
			'248': {
				linkType: 'page',
				data: {
					selected: [{ id: '12345' }],
				},
			},
		};

		const tree = require('react-test-renderer').create(
			<AtomLinkPicker
				{...defaultProps}
				links={links}
			/>
		).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should render correctly for brand', () => {
		const links = {
			'248': {
				linkType: 'brand',
				data: {
					selected: [{ id: '12345' }],
				},
			},
		};

		const tree = require('react-test-renderer').create(
			<AtomLinkPicker
				{...defaultProps}
				links={links}
			/>
		).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
