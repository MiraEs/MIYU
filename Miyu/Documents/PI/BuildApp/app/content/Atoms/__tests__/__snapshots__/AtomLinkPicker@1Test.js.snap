// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtomLinkPicker component should render correctly 1`] = `null`;

exports[`AtomLinkPicker component should render correctly for article 1`] = `
<TouchableOpacity
  activeOpacity={0.5}
  onPress={[Function]}
  trackAction="trackAction"
  trackContextData={
    Object {
      "_a": "12345",
      "_cr": "contentItemId",
      "_g": 1,
    }
  }
/>
`;

exports[`AtomLinkPicker component should render correctly for brand 1`] = `
<TouchableOpacity
  activeOpacity={0.5}
  onPress={[Function]}
  trackAction="trackAction"
  trackContextData={
    Object {
      "_b": "12345",
      "_cr": "contentItemId",
      "_g": 1,
    }
  }
/>
`;

exports[`AtomLinkPicker component should render correctly for category 1`] = `
<TouchableOpacity
  activeOpacity={0.5}
  onPress={[Function]}
  trackAction="trackAction"
  trackContextData={
    Object {
      "_c": "123456",
      "_cr": "contentItemId",
      "_g": 1,
    }
  }
/>
`;

exports[`AtomLinkPicker component should render correctly for favorite 1`] = `
<TouchableOpacity
  activeOpacity={0.5}
  onPress={[Function]}
  trackAction="trackAction"
  trackContextData={
    Object {
      "_cr": "contentItemId",
      "_f": "12345",
      "_g": 1,
    }
  }
/>
`;

exports[`AtomLinkPicker component should render correctly for page 1`] = `
<TouchableOpacity
  activeOpacity={0.5}
  onPress={[Function]}
  trackAction="trackAction"
  trackContextData={
    Object {
      "_cr": "contentItemId",
      "_g": 1,
      "_r": "12345",
    }
  }
/>
`;

exports[`AtomLinkPicker component should render correctly for product 1`] = `
<TouchableOpacity
  activeOpacity={0.5}
  onPress={[Function]}
  trackAction="trackAction"
  trackContextData={
    Object {
      "_cr": "contentItemId",
      "_g": 1,
      "_s": "12345",
    }
  }
/>
`;

exports[`AtomLinkPicker component should render correctly for profile 1`] = `
<TouchableOpacity
  activeOpacity={0.5}
  onPress={[Function]}
  trackAction="trackAction"
  trackContextData={
    Object {
      "_cr": "contentItemId",
      "_g": 1,
    }
  }
/>
`;

exports[`AtomLinkPicker component should render correctly for video 1`] = `
<TouchableOpacity
  activeOpacity={0.5}
  onPress={[Function]}
  trackAction="trackAction"
  trackContextData={
    Object {
      "_cr": "contentItemId",
      "_g": 1,
    }
  }
/>
`;
